File: src/modifiers/italic.ts
import { createModifier } from './createModifier';

export interface ItalicParams {
  /// Whether italic is active. Defaults to `true`.
  value?: boolean;
}

/// SwiftUI `.italic(_:)`. Applies italics to text within the view. Pass
/// `false` to explicitly turn it off.
export function italic(params?: ItalicParams | boolean) {
  if (typeof params === 'boolean') {
    return createModifier('italic', { value: params });
  }
  return createModifier('italic', params);
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/Italic.swift
import SwiftUI

/// SwiftUI `.italic(_:)`. Applies italics to text within the view.
/// Defaults to active when `value` is omitted.
enum RNWItalicModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("italic") { view, params, _ in
            AnyView(view.italic(params.bool("value") ?? true))
        }
    }
}
